.github/workflows/ci.yml

```diff
@@ -0,0 +1,177 @@
+name: CI
+
+on:
+    push:
+    pull_request:
+
+jobs:
+    php-cs-fixer:
+        runs-on: ubuntu-latest
+        name: Coding Standards
+        steps:
+            - name: Checkout
+              uses: actions/checkout@v2
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: '7.4'
+                  tools: php-cs-fixer, cs2pr
+
+            - name: PHP Coding Standards Fixer
+              run: php-cs-fixer fix --dry-run --format checkstyle | cs2pr
+
+    phpstan:
+        runs-on: ubuntu-latest
+        name: Static Analysis
+        steps:
+            -   name: Checkout
+                uses: actions/checkout@v2
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: '7.4'
+                  tools: phpstan, cs2pr
+
+            - name: Get composer cache directory
+              id: composercache
+              run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            - name: Cache dependencies
+              uses: actions/cache@v2
+              with:
+                  path: ${{ steps.composercache.outputs.dir }}
+                  key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
+                  restore-keys: ${{ runner.os }}-composer-
+
+            - name: Install dependencies
+              run: composer install --prefer-dist
+
+            - name: Install PHPUnit dependencies
+              run: vendor/bin/simple-phpunit --version
+
+            - name: PHPStan
+              run: phpstan analyse --no-progress --error-format=checkstyle | cs2pr
+
+    phpunit:
+        runs-on: ubuntu-latest
+        strategy:
+            matrix:
+                php-versions: ['7.1', '7.2', '7.3', '7.4']
```

Comment:
I suggest adding `fail-fast: false` in the strategy too, so that tests run on the 4 PHP versions all the time, rather than being cancelled when one of the fail